server.js

```diff
@@ -1,53 +1,86 @@
-		res.write('Request body:\n');
-		res.write(util.inspect(formattedBody, null) + '\n');
-		res.write('\n');
-		res.write('Host: ' + os.hostname() + '\n');
-		res.write('OS Type: ' + os.type() + '\n');
-		res.write('OS Platform: ' + os.platform() + '\n');
-		res.write('OS Arch: ' + os.arch() + '\n');
-		res.write('OS Release: ' + os.release() + '\n');
-		res.write('OS Uptime: ' + os.uptime() + '\n');
-		res.write('OS Free memory: ' + os.freemem() / 1024 / 1024 + 'mb\n');
-		res.write('OS Total memory: ' + os.totalmem() / 1024 / 1024 + 'mb\n');
-		res.write('OS CPU count: ' + os.cpus().length + '\n');
-		res.write('OS CPU model: ' + os.cpus()[0].model + '\n');
-		res.write('OS CPU speed: ' + os.cpus()[0].speed + 'mhz\n');
-		res.end('\n');
-	});
-	next();
+var fs      = require('fs');
+var app     = express();
+var eps     = require('ejs');
+
+app.engine('html', require('ejs').renderFile);
+
+var port = process.env.PORT || process.env.OPENSHIFT_NODEJS_PORT || 8080;
+var ip   = process.env.IP   || process.env.OPENSHIFT_NODEJS_IP || '0.0.0.0';
+var mongoURL = process.env.OPENSHIFT_MONGODB_DB_URL || process.env.MONGO_URL;
+var mongoURLLabel = "";
+if (mongoURL == null && process.env.DATABASE_SERVICE_NAME) {
+  var mongoServiceName = process.env.DATABASE_SERVICE_NAME.toUpperCase();
+  var mongoHost = process.env[mongoServiceName + "_SERVICE_HOST"];
+  var mongoPort = process.env[mongoServiceName + "_SERVICE_PORT"];
+  var mongoUser = process.env.MONGODB_USER
+  if (mongoHost && mongoPort && process.env.MONGODB_DATABASE) {
+    mongoURLLabel = mongoURL = 'mongodb://';
+    if (process.env.MONGODB_USER && process.env.MONGODB_PASSWORD) {
+      mongoURL += process.env.MONGODB_USER + ':' + process.env.MONGODB_PASSWORD + '@';
+    }
+    // Provide UI label that excludes user id and pw
+
+    mongoURLLabel += mongoHost + ':' + mongoPort + '/' + process.env.MONGODB_DATABASE;
+    mongoURL += mongoHost + ':' + mongoPort + '/' + process.env.MONGODB_DATABASE;
+  }
+}
+var db = null;
+var dbDetails = new Object();
+
+var initDb = function(callback) {
+  if (mongoURL == null) return;
+
+  var mongodb = require('mongodb');  
+  if (mongodb == null) return;
+
+  mongodb.connect(mongoURL, function(err, conn) {
+    if (err) {
+      callback(err);
+      return;
+    }
+
+    db = conn;
+    dbDetails.databaseName = db.databaseName;
+    dbDetails.url = mongoURLLabel;
+    dbDetails.type = 'MongoDB';
+
+    console.log("Connected to MongoDB at: " + mongoURL);
+  });
+};
+
+app.get('/', function (req, res) {
+  if (db) {
+    var col = db.collection('counts');
+    // Create a document with request IP and current time of request
+    col.insert({ip: req.ip, date: Date.now()});
+    col.count(function(err, count){
+      res.render('index.html', { pageCountMessage : count, dbInfo: dbDetails });
+    });
+  } else {
+    res.render('index.html', { pageCountMessage : null});
+  }
+});
+
+app.get('/pagecount', function (req, res) {
```

Comment:
I curl it sometimes to test, does it hurt anything by having it? I was thinking of doing a polling & websockets add-on sample, so may use it there as well (polling)